### ISSUE TITLE
IllegalStateException when groupByCommit called after compare

### ISSUE THREAD

**Issue by swhittaker** [javers/javers#1160]
**After a compare I expected to be able to call pretty print on the Changes, instead an exception is seen**
...

**Steps To Reproduce**
I have a runnable test case which isolates the bug and allows Javers Core Team to easily reproduce it.
It is tiny so included below:

```
    public static class MyClass{
        String value;
        public MyClass(String value){
            this.value = value;
        }
    }

    public static void main(String[] args) {
        final Javers javers = JaversBuilder.javers().build();
        final Diff diff = javers.compare(new MyClass("abc"), new MyClass("cde"));
        diff.getChanges().prettyPrint();
    }
```

**Javers' Version**
Tested on 6.4.1 and 5.14.0

**Comment by bartoszwalacik** [comment]
thanks for reporting

**Comment by bartoszwalacik** [comment]
fixed in 6.5.2
